lightning/src/ln/channelmanager.rs

```diff
@@ -7812,26 +7771,38 @@ where
 						return Err(MsgHandleErrInternal::from_chan_no_close(e, msg.common_fields.temporary_channel_id));
 					},
 					Ok(res) => res
-				};
+				})
+			},
+		};
 
-				let channel_type = channel.context.get_channel_type();
-				if channel_type.requires_zero_conf() {
-					return Err(MsgHandleErrInternal::send_err_msg_no_close("No zero confirmation channels accepted".to_owned(), msg.common_fields.temporary_channel_id));
-				}
-				if channel_type.requires_anchors_zero_fee_htlc_tx() {
-					return Err(MsgHandleErrInternal::send_err_msg_no_close("No channels with anchor outputs accepted".to_owned(), msg.common_fields.temporary_channel_id));
-				}
+		let channel_type = channel_phase.context().get_channel_type();
+		if channel_type.requires_zero_conf() {
+			return Err(MsgHandleErrInternal::send_err_msg_no_close("No zero confirmation channels accepted".to_owned(), common_fields.temporary_channel_id));
+		}
+		if channel_type.requires_anchors_zero_fee_htlc_tx() {
+			return Err(MsgHandleErrInternal::send_err_msg_no_close("No channels with anchor outputs accepted".to_owned(), common_fields.temporary_channel_id));
+		}
 
-				let outbound_scid_alias = self.create_and_insert_outbound_scid_alias();
-				channel.context.set_outbound_scid_alias(outbound_scid_alias);
+		let outbound_scid_alias = self.create_and_insert_outbound_scid_alias();
+		channel_phase.context_mut().set_outbound_scid_alias(outbound_scid_alias);
 
-				peer_state.pending_msg_events.push(events::MessageSendEvent::SendAcceptChannelV2 {
-					node_id: *counterparty_node_id,
-					msg: channel.accept_inbound_dual_funded_channel(),
-				});
-				peer_state.channel_by_id.insert(channel.context.channel_id(), ChannelPhase::UnfundedInboundV2(channel));
+		let message_send_event = match channel_phase {
+			ChannelPhase::UnfundedInboundV1(ref mut channel) => events::MessageSendEvent::SendAcceptChannel {
+				node_id: *counterparty_node_id,
+				msg: channel.accept_inbound_channel(),
 			},
-		}
+			ChannelPhase::UnfundedInboundV2(ref mut channel) => events::MessageSendEvent::SendAcceptChannelV2 {
+				node_id: *counterparty_node_id,
+				msg: channel.accept_inbound_dual_funded_channel(),
+			},
+			_ => {
+				debug_assert!(false, "We can only have an unfunded inbound V1/V2 channel here");
+				return Err(MsgHandleErrInternal::send_err_msg_no_close("This should be an inbound channel.".to_owned(), common_fields.temporary_channel_id))
+			}
```

Comment:
We can get rid of this branch with the following rearrangement. Though note that the `MessageSendEvent` is created prior to the checks made against the channel type that are mentioned in another comment.

```diff
diff --git a/lightning/src/ln/channel.rs b/lightning/src/ln/channel.rs
index 20810b000..4e9bcb6e2 100644
--- a/lightning/src/ln/channel.rs
+++ b/lightning/src/ln/channel.rs
@@ -8504,7 +8504,7 @@ impl<SP: Deref> InboundV1Channel<SP> where SP::Target: SignerProvider {
        /// should be sent back to the counterparty node.
        ///
        /// [`msgs::AcceptChannel`]: crate::ln::msgs::AcceptChannel
-       pub fn accept_inbound_channel(&mut self) -> msgs::AcceptChannel {
+       pub fn accept_inbound_channel(&self) -> msgs::AcceptChannel {
                if self.context.is_outbound() {
                        panic!("Tried to send accept_channel for an outbound channel?");
                }
@@ -8881,7 +8881,7 @@ impl<SP: Deref> InboundV2Channel<SP> where SP::Target: SignerProvider {
        /// should be sent back to the counterparty node.
        ///
        /// [`msgs::AcceptChannelV2`]: crate::ln::msgs::AcceptChannelV2
-       pub fn accept_inbound_dual_funded_channel(&mut self) -> msgs::AcceptChannelV2 {
+       pub fn accept_inbound_dual_funded_channel(&self) -> msgs::AcceptChannelV2 {
                if self.context.is_outbound() {
                        debug_assert!(false, "Tried to send accept_channel for an outbound channel?");
                }
diff --git a/lightning/src/ln/channelmanager.rs b/lightning/src/ln/channelmanager.rs
index 55ac16802..7c3f10f4c 100644
--- a/lightning/src/ln/channelmanager.rs
+++ b/lightning/src/ln/channelmanager.rs
@@ -7767,30 +7767,32 @@ where
                random_bytes.copy_from_slice(&self.entropy_source.get_secure_random_bytes()[..16]);
                let user_channel_id = u128::from_be_bytes(random_bytes);
 
-               let mut channel_phase = match msg {
+               let (mut channel_phase, message_send_event) = match msg {
                        OpenChannelMessageRef::V1(msg) => {
-                               ChannelPhase::UnfundedInboundV1(match InboundV1Channel::new(
+                               let channel = InboundV1Channel::new(
                                        &self.fee_estimator, &self.entropy_source, &self.signer_provider, *counterparty_node_id,
                                        &self.channel_type_features(), &peer_state.latest_features, msg, user_channel_id,
                                        &self.default_configuration, best_block_height, &self.logger, /*is_0conf=*/false
-                               ) {
-                                       Err(e) => {
-                                               return Err(MsgHandleErrInternal::from_chan_no_close(e, msg.common_fields.temporary_channel_id));
-                                       },
-                                       Ok(res) => res
-                               })
+                               )
+                                       .map_err(|e| MsgHandleErrInternal::from_chan_no_close(e, msg.common_fields.temporary_channel_id))?;
+                               let message_send_event = events::MessageSendEvent::SendAcceptChannel {
+                                       node_id: *counterparty_node_id,
+                                       msg: channel.accept_inbound_channel(),
+                               };
+                               (ChannelPhase::UnfundedInboundV1(channel), message_send_event)
                        },
                        OpenChannelMessageRef::V2(msg) => {
-                               ChannelPhase::UnfundedInboundV2(match InboundV2Channel::new(&self.fee_estimator, &self.entropy_source,
-                                       &self.signer_provider, self.get_our_node_id(), *counterparty_node_id, &self.channel_type_features(),
-                                       &peer_state.latest_features, msg, vec![], user_channel_id, &self.default_configuration,
-                                       best_block_height, &self.logger)
-                               {
-                                       Err(e) => {
-                                               return Err(MsgHandleErrInternal::from_chan_no_close(e, msg.common_fields.temporary_channel_id));
-                                       },
-                                       Ok(res) => res
-                               })
+                               let channel = InboundV2Channel::new(
+                                       &self.fee_estimator, &self.entropy_source, &self.signer_provider, self.get_our_node_id(),
+                                       *counterparty_node_id, &self.channel_type_features(), &peer_state.latest_features, msg,
+                                       vec![], user_channel_id, &self.default_configuration, best_block_height, &self.logger
+                               )
+                                       .map_err(|e| MsgHandleErrInternal::from_chan_no_close(e, msg.common_fields.temporary_channel_id))?;
+                               let message_send_event = events::MessageSendEvent::SendAcceptChannelV2 {
+                                       node_id: *counterparty_node_id,
+                                       msg: channel.accept_inbound_dual_funded_channel(),
+                               };
+                               (ChannelPhase::UnfundedInboundV2(channel), message_send_event)
                        },
                };
 
@@ -7805,20 +7807,6 @@ where
                let outbound_scid_alias = self.create_and_insert_outbound_scid_alias();
                channel_phase.context_mut().set_outbound_scid_alias(outbound_scid_alias);
 
-               let message_send_event = match channel_phase {
-                       ChannelPhase::UnfundedInboundV1(ref mut channel) => events::MessageSendEvent::SendAcceptChannel {
-                               node_id: *counterparty_node_id,
-                               msg: channel.accept_inbound_channel(),
-                       },
-                       ChannelPhase::UnfundedInboundV2(ref mut channel) => events::MessageSendEvent::SendAcceptChannelV2 {
-                               node_id: *counterparty_node_id,
-                               msg: channel.accept_inbound_dual_funded_channel(),
-                       },
-                       _ => {
-                               debug_assert!(false, "We can only have an unfunded inbound V1/V2 channel here");
-                               return Err(MsgHandleErrInternal::send_err_msg_no_close("This should be an inbound channel.".to_owned(), common_fields.temporary_channel_id))
-                       }
-               };
                peer_state.pending_msg_events.push(message_send_event);
                peer_state.channel_by_id.insert(channel_phase.context().channel_id(), channel_phase);
```